tiltak/src/main/kotlin/no/nav/amt/tiltak/deltaker/service/EndringsmeldingService.kt

```diff
@@ -0,0 +1,26 @@
+package no.nav.amt.tiltak.deltaker.service
+
+import no.nav.amt.tiltak.deltaker.dbo.EndringsmeldingDbo
+import no.nav.amt.tiltak.deltaker.repositories.EndringsmeldingRepository
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.EnableTransactionManagement
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+import java.util.*
+
+@Service
+@EnableTransactionManagement
+open class EndringsmeldingService(
+	private val repository: EndringsmeldingRepository,
+) {
+
+	fun opprettMedStartDato(deltakerId: UUID, startDato: LocalDate, ansattId: UUID) {
+		insertOgInaktiverStartDato(startDato, deltakerId, ansattId)
+	}
+
+	@Transactional
+	open fun insertOgInaktiverStartDato(startDato: LocalDate, deltakerId: UUID, opprettetAv: UUID): EndringsmeldingDbo {
+		repository.inaktiverTidligereMeldinger(deltakerId)
+		return repository.insertNyStartDato(startDato, deltakerId, opprettetAv)
+	}
+}
```

Comment:
`opprettMedStartDato()` vil bli kjørt uten transaksjon siden "@"Transactional kun fungerer når man kaller på metoden fra utsiden av klassen (grunnet hvordan spring setter opp AOP proxyen).

Anbefaler heller å bruke `private val transactionTemplate: TransactionTemplate,` og `transactionTemplate.executeWithoutResult {...}` istedenfor. transactionTemplate blir automatisk laget av Spring, så man kan injecte den i konstruktøren uten å opprette egen bønne

Reply:
Hmm, virker som jeg egentlig bare burde ha den i repository tenker jeg da. Ønsket i utgangspunktet ikke å eksponere de to metodene ut av repository uansett fordi de per nå ikke skal brukes separat

Reply:
Det kommer litt på hva slags ansvar man ønsker at repository skal ha. Personlig så tenker jeg at repositories skal være dumme og kun gjøre enkle ting, mens at alt av logikk ligger i services.